lalrpop-util/src/lib.rs

```diff
@@ -108,14 +118,32 @@ impl<L, T, E> ParseError<L, T, E> {
         }
     }
 
+    /// Transform a ParseError by applying a function to the location field
+    ///
+    /// This could be useful for example to obtain a `ParseError` where all
+    /// fields implement `Display` or `Debug`.  `ParseErrors` where all fields
+    /// implement `Display` also implement `Display` and `ParseErrors` where all
+    /// fields implement both `Display` and `Debug` implement `Error`
     pub fn map_location<LL>(self, op: impl FnMut(L) -> LL) -> ParseError<LL, T, E> {
         self.map_intern(op, |x| x, |x| x)
     }
 
+    /// Transform a ParseError by applying a function to the token field
+    ///
+    /// This could be useful for example to obtain a `ParseError` where all
+    /// fields implement `Display` or `Debug`.  `ParseErrors` where all fields
+    /// implement `Display` also implement `Display` and `ParseErrors` where all
+    /// fields implement both `Display` and `Debug` implement `Error`
     pub fn map_token<TT>(self, op: impl FnOnce(T) -> TT) -> ParseError<L, TT, E> {
         self.map_intern(|x| x, op, |x| x)
     }
 
+    /// Transform a ParseError by applying a function to the error field
+    ///
+    /// This could be useful for example to obtain a `ParseError` where all
+    /// fields implement `Display` or `Debug`.  `ParseErrors` where all fields
+    /// implement `Display` also implement `Display` and `ParseErrors` where all
+    /// fields implement both `Display` and `Debug` implement `Error`
```

Comment:
I'm not sure that this example is clear. Basically what we are providing is https://doc.rust-lang.org/std/result/enum.Result.html#method.map_err but for Lalrpop error construct.

Presumably these functions are more general. Like applying an offset value to a location or mapping a token character to it's escaped version.

Reply:
I reworded these to try to make things more clear.  What do you think of the new version?